e2e/searchCommits_test.go

```diff
@@ -36,5 +36,21 @@ var _ = Describe("E2E TEST:SearchCommits", func() {
 			Expect(err).Should(BeNil())
 			Expect(len(commits)).Should(BeNumerically(">", 0))
 		})
+		It("Should return error as it is not using HEAD", func() {
+			repo, err := githubrepo.MakeGithubRepo("ossf/scorecard")
```

Comment:
Can we use a smaller repo from `ossf-tests` perhaps? Every now and then I go back and prune these references since downloading `ossf/scorecard` takes some time.

Reply:
The previous tests require a valid repo with dependabot, and so these tests would cache the existing repo and shouldn't be an issue.

Reply:
Caching is done at the repoclient level right? And there's no repo client shared between these tests (and it can't be since one expects an err and the other doesnt)

Reply:
I'm actually wrong here for another reason: these tests don't trigger the tarball download.